docs/5.0/pages/database-access/architecture.mdx

```diff
@@ -0,0 +1,126 @@
+* [Teleport Proxy](../architecture/proxy.mdx). It's a stateless service
+  that performs a function of an authentication gateway, serves web UI and
+  accepts database (and other) client connections.
+* [Teleport Auth](../architecture/authentication.mdx). It serves as
+  cluster's certificate authority, handles user authentication/authorization
+  and issues short-lived client certificates.
+* Teleport Database Service. It's the Database Access' "brain" that connects
+  to the databases, performs database authentication and protocol parsing.
+
+To connect back to the cluster, Database Service establishes an SSH reverse
+tunnel to the Proxy.
+
+As such, users do not need to have direct connectivity to the Database Service,
+or the databases it's connected to. As long as it can dial back to the cluster's
+Proxy server, it can be located behind a firewall.
+
+<Admonition type="tip" title="Tip">
+You can have multiple Database Services connected to the cluster, and each
+Database Service can be connected to multiple databases.
+</Admonition>
+
+Let's take a look at the typical flow Database Access users go through to
+connect to a database.
+
+1. First, a user logs into the cluster with `tsh login` command and retrieves
+   a client certificate. See [Issuing User Certificates](../architecture/authentication.mdx#issuing-user-certificates)
+   for more details on how it works.
+2. Once logged in, the user picks the database they want to connect to from
+   the list of available databases shown in `tsh db ls` command and retrieves
+   a short-lived certificate for it with `tsh db login`.
+3. The user uses a standard database client (e.g. `psql`, `mysql` or one of the
+   [graphical clients](./guides/gui-clients.mdx)) to connect to the Proxy, authenticating
+   with the client certificate from step 2.
+4. The Proxy authenticates the connection and dispatches it to the appropriate
+   Database Service based on the routing information encoded in the client
+   certificate, over the reverse tunnel.
+5. The Database Service also authenticates the connection, does an authorization
+   check to make sure the user has access to the database they's connecting to
+   and establishes the connection to the database.
+6. Once the connection has been established, the Database Service starts proxying traffic
+   between the user's database client and the database and interpreting the
+   database wire protocol messages to submit events to the audit log.
+
+## Authentication
+
+Teleport relies on short-lived client certificates as the primary means of
+authentication, both for user authentication as well as authentication between
+internal components.
+
+In the Database Access architecture, authentication happens in 3 places:
+database client to Proxy, Proxy to Database Service, and Database Service to
+database. Let's take a look in more detail at each one.
+
+### Database Client to Proxy
+
+Database clients authenticate with the Proxy using x509 client certificates
```

Comment:
It might be good to explain that many database clients refer this these certificates as 'SSL' certs, but this is just legacy terminology